main.js

```diff
@@ -34,6 +34,9 @@ module.exports = class HotReload extends Plugin {
     watch(path) {
         if (this.app.vault.adapter.watchers.hasOwnProperty(path)) return;
         const realPath = [this.app.vault.adapter.basePath, path].join("/");
+        if (!fs.existsSync(realPath)) {
+            return;
+        }
```

Comment:
Rather than checking for the existence, it's simpler to add `, {throwIfNoEntry: false}` to the lstatSync call.

Reply:
you are very "jsnic"
I've just did 2 more changes. I changed the notice message because that too long on screen. I just put "hot reloaded" because I know the name of the plugin I'm editing. And I put a longer timer on reload, to have the time to do a change, get back to obsidian. But maybe because I have only one screen.
I think it would maybe possible to check on start of hotreload if one of tracked files has changed. because on my plugin quick plugin switcher I can a add a command to any plugin to switch it. I guess this is what your command is doing. So I will maybe try to put it onload.  Just to share...